src/javascript/binary/socket.js

```diff
@@ -62,16 +62,18 @@ var BinarySocket = (function () {
             if(!data.hasOwnProperty('passthrough')){
                 data.passthrough = {};
             }
-            data.passthrough.req_number = ++req_number;
-            timeouts[req_number] = setInterval(function(){
-                if(typeof reloadPage === 'function'){
-                    var r = confirm("The server didn't respond. Reload page?");
-                    console.log(data,'Last Error Request');
-                    if (r === true) {
-                        reloadPage();
-                    } 
-                }
-            }, 7*1000);
+            if(data.contracts_for || data.proposal){
+                data.passthrough.req_number = ++req_number;
+                timeouts[req_number] = setInterval(function(){
+                    if(typeof reloadPage === 'function'){
+                        var r = confirm("The server didn't respond to the request:\n\n"+JSON.stringify(data)+"\n\nReload page?");
+                        if (r === true) {
+                            reloadPage();
+                        } 
+                    }
+                }, 7*1000);
```

Comment:
Multiple issues IMHO:
 1. Why `setInterval`? We should just `setTimeout` here and then no need to `clearTimeouts`.
 2. Other UIs increment the next check's interval each time it fails (e.g. Gmail first retries in 5 seconds, if that fails it then tries in 15 seconds, ...). Here we do it every time after 7 seconds.
 3. Other UI's do not present a `confirm`, they show a disconnected message (a `div`) and a count-down timer. On timeout, they do retry (to re-connect, not relaod). That is superior.